src/app/rundown-view/components/minishelf/mini-shelf.component.spec.ts

```diff
@@ -1,28 +1,205 @@
 import { ComponentFixture, TestBed } from '@angular/core/testing'
 import { MiniShelfComponent } from './mini-shelf.component'
 import { ConfigurationService } from '../../../shared/services/configuration.service'
-import { instance, mock } from '@typestrong/ts-mockito'
+import { anyString, instance, mock, verify, when } from '@typestrong/ts-mockito'
 import { ActionStateService } from '../../../shared/services/action-state.service'
 import { TimerPipe } from '../../../shared/pipes/timer/timer.pipe'
 import { MediaStateService } from '../../../shared/services/media-state.service'
 import { ActionService } from '../../../shared/abstractions/action.service'
 import { Logger } from '../../../core/abstractions/logger.service'
+import { TestEntityFactory } from '../../../test/factories/test-entity.factory'
+import { StudioConfiguration } from '../../../shared/models/studio-configuration'
+import { Observable } from 'rxjs'
+import { Media } from '../../../shared/services/media'
+import { Tv2SegmentMetadata } from '../../../core/models/tv2-segment-metadata'
 
 describe('MiniShelfComponent', () => {
-  it('should create', async () => {
-    const component: MiniShelfComponent = await configureTestBed()
-    expect(component).toBeTruthy()
+  it('should have segment name capitalized as title text', async () => {
+    await configureTestBed({
+      mockedConfigurationService: createConfigurationService(),
+    })
+
+    const fixture: ComponentFixture<MiniShelfComponent> = TestBed.createComponent(MiniShelfComponent)
+    const component: MiniShelfComponent = fixture.componentInstance
+    const testEntityFactory: TestEntityFactory = new TestEntityFactory()
+    const segmentName: string = 'segment-with-video-clip'
+    component.segment = testEntityFactory.createSegment({ name: segmentName })
+    component.videoClipAction = testEntityFactory.createTv2VideoClipAction()
+
+    fixture.detectChanges()
+
+    const titleElement: HTMLElement = fixture.nativeElement.querySelector('span.c-mini-shelf__title')
+    const expectedTitle: string = segmentName.toUpperCase()
+    expect(titleElement?.textContent).toBe(expectedTitle)
+  })
+
+  it('should have correct value for calculated duration', async () => {
```

Comment:
I see you did the change :)

When writing tests like this it is not enough to just have a single test for it because I can just hardcode the expected return value of the single test and everything passes, but my logic would be incorrect (see image).
![image](https://github.com/tv2/ng-sofie/assets/23049647/fea47581-abd9-4954-a9d9-8c6c0d2432bb)

So when writing tests like this, a good number of tests would be three tests (with different values) + any edgecase we might find/deem necesary.